app/services/members/delete_service.rb

```diff
@@ -60,4 +60,13 @@ def cleanup_for_group(member)
       .new(member.principal, current_user: user, contract_class: EmptyContract)
       .call
   end
+
+  def default_contract_class
+    # We have different contracts for project roles and global roles
+    if model.project.present?
+      "#{namespace}::DeleteContract".constantize
+    else
+      "#{namespace}::GlobalDeleteContract".constantize
+    end
```

Comment:
A [spec for this](https://github.com/opf/openproject/blob/dev/spec/contracts/members/delete_contract_spec.rb) would ensure that the behaviour is safe against regression.

Reply:
added with 2e4b0eb70f30